ts/sagas/wallet.ts

```diff
@@ -754,20 +754,18 @@ export function* watchBackToEntrypointPaymentSaga(): Iterator<Effect> {
       GlobalState
     >(_ => _.wallet.payment.entrypointRoute);
     if (entrypointRoute !== undefined) {
-      const routeName = entrypointRoute ? entrypointRoute.name : undefined;
       const key = entrypointRoute ? entrypointRoute.key : undefined;
-
-      // if the payment was initiated from the message list or detail go back
-      if (
-        routeName === ROUTES.MESSAGES_HOME ||
-        routeName === ROUTES.MESSAGE_DETAIL
-      ) {
-        const navigationBackAction = NavigationActions.back({
-          key
-        });
-        yield put(navigationBackAction);
-      } else {
-        yield put(navigateToWalletHome());
+      const routeName = entrypointRoute ? entrypointRoute.name : undefined;
+      const navigationBackAction = NavigationActions.back({
+        key
+      });
+      yield put(navigationBackAction);
```

Comment:
are **navigateBack** and **navigationBackAction** the same actions?

Reply:
Yes they dispatch the same action but with the body key as a payload, i think we can switch with the action creator